javascript/example/DRACOLoader.js

```diff
@@ -346,131 +333,109 @@ THREE.DRACOLoader.prototype = {
+  THREE.DRACOLoader.decoderPath = path;
+};
 
 /**
- * Creates and returns a singleton instance of the DracoDecoderModule.
- * The module loading is done asynchronously for WebAssembly. Initialized module
- * can be accessed through the callback function
- * |onDracoDecoderModuleLoadedCallback|.
+ * Sets decoder configuration and releases singleton decoder module. Module
+ * will be recreated with the next decoding call.
+ * @param {Object} config
  */
-THREE.DRACOLoader.getDecoderModule = (function() {
-    return function(onDracoDecoderModuleLoadedCallback) {
-        if (typeof THREE.DRACOLoader.decoderModule !== 'undefined') {
-          // Module already initialized.
-          if (typeof onDracoDecoderModuleLoadedCallback !== 'undefined') {
-            onDracoDecoderModuleLoadedCallback(THREE.DRACOLoader.decoderModule);
-          }
-        } else {
-          if (typeof DracoDecoderModule === 'undefined') {
-            // Wait until the Draco decoder is loaded before starting the error
-            // timer.
-            if (THREE.DRACOLoader.timeLoaded > 0) {
-              var waitMs = performance.now() - THREE.DRACOLoader.timeLoaded;
-
-              // After loading the Draco JavaScript decoder file, there is still
-              // some time before the DracoDecoderModule is defined. So start a
-              // loop to check when the DracoDecoderModule gets defined. If the
-              // time is hit throw an error.
-              if (waitMs > 5000) {
-                throw new Error(
-                    'THREE.DRACOLoader: DracoDecoderModule not found.');
-              }
-            }
-          } else {
-            if (!THREE.DRACOLoader.decoderCreationCalled) {
-              THREE.DRACOLoader.decoderCreationCalled = true;
-              THREE.DRACOLoader.dracoDecoderType['onModuleLoaded'] =
-                  function(module) {
-                    THREE.DRACOLoader.decoderModule = module;
-                  };
-              DracoDecoderModule(THREE.DRACOLoader.dracoDecoderType);
-            }
-          }
+THREE.DRACOLoader.setDecoderConfig = function ( config ) {
+  var wasmBinary = THREE.DRACOLoader.decoderConfig.wasmBinary;
+  THREE.DRACOLoader.decoderConfig = config || {};
+  THREE.DRACOLoader.decoderModulePromise = null;
 
-          // Either the DracoDecoderModule has not been defined or the decoder
-          // has not been created yet. Call getDecoderModule() again.
-          setTimeout(function() {
-            THREE.DRACOLoader.getDecoderModule(
-                onDracoDecoderModuleLoadedCallback);
-          }, 10);
-        }
-    };
+  // Reuse WASM binary.
+  if ( wasmBinary ) THREE.DRACOLoader.decoderConfig.wasmBinary = wasmBinary;
+};
 
-})();
-
-// Releases the DracoDecoderModule instance associated with the draco loader.
-// The module will be automatically re-created the next time a new geometry is
-// loaded by THREE.DRACOLoader.
-THREE.DRACOLoader.releaseDecoderModule = function() {
```

Comment:
is there a reason why this got removed?

Reply:
Releasing now happens automatically if you call `DRACOLoader.setDecoderConfig({...})` and I assumed it wasn't needed otherwise, but could add an explicit method back if that's preferred.

Reply:
From https://github.com/google/draco/issues/312 I guess we do want to keep the method to explicitly release resources?

Reply:
I think we should keep it there. Otherwise other changes look good to us.

Reply:
Ok, sounds good to me. The `.releaseDecoderModule()` method is back again.